packages/core/test/new-api/readEventArgument.ts

```diff
@@ -0,0 +1,211 @@
+      const fakeArtifact = {} as any;
+
+      const defintion = defineModule("Module1", (m) => {
+        const contract = m.contract("Contract");
+        const contractFromArtifact = m.contractFromArtifact(
+          "ContractFromArtifact",
+          fakeArtifact
+        );
+        const call = m.call(contract, "fuc");
+
+        m.readEventArgument(contract, "EventName1", "arg1");
+        m.readEventArgument(contractFromArtifact, "EventName2", "arg2");
+        m.readEventArgument(call, "EventName3", "arg3");
+
+        return { contract, contractFromArtifact };
+      });
+
+      const constructor = new ModuleConstructor(0, []);
+      const mod = constructor.construct(defintion);
+
+      const callFuture = Array.from(mod.futures).find(
+        (f) => f.type === FutureType.NAMED_CONTRACT_CALL
+      );
+
+      const [read1, read2, read3] = Array.from(mod.futures).filter(
+        (f) => f.type === FutureType.READ_EVENT_ARGUMENT
+      ) as ReadEventArgumentFuture[];
+
+      assert.equal(read1.futureToReadFrom, mod.results.contract);
+      assert.equal(read2.futureToReadFrom, mod.results.contractFromArtifact);
+      assert.equal(read3.futureToReadFrom, callFuture);
+    });
+
+    it("Should infer the emitter from the future correctly", () => {
+      const defintion = defineModule("Module1", (m) => {
+        const contract = m.contract("Contract");
+        const call = m.call(contract, "fuc");
+
+        m.readEventArgument(contract, "EventName1", "arg1");
+        m.readEventArgument(call, "EventName2", "arg2");
+
+        return { contract };
+      });
+
+      const constructor = new ModuleConstructor(0, []);
+      const mod = constructor.construct(defintion);
+
+      const [read1, read2] = Array.from(mod.futures).filter(
+        (f) => f.type === FutureType.READ_EVENT_ARGUMENT
+      ) as ReadEventArgumentFuture[];
+
+      assert.equal(read1.emitter, mod.results.contract);
+      assert.equal(read2.emitter, mod.results.contract);
+    });
+
+    it("Should accept an explicit emitter", () => {
+      const defintion = defineModule("Module1", (m) => {
+        const contract = m.contract("Contract");
+        const call = m.call(contract, "fuc");
+
+        const contract2 = m.contract("Contract2");
+
+        m.readEventArgument(contract, "EventName1", "arg1");
+        m.readEventArgument(call, "EventName2", "arg2", { emitter: contract2 });
+
+        return { contract, contract2 };
+      });
+
+      const constructor = new ModuleConstructor(0, []);
+      const mod = constructor.construct(defintion);
+
+      const [read1, read2] = Array.from(mod.futures).filter(
+        (f) => f.type === FutureType.READ_EVENT_ARGUMENT
+      ) as ReadEventArgumentFuture[];
+
+      assert.equal(read1.emitter, mod.results.contract);
+      assert.equal(read2.emitter, mod.results.contract2);
+    });
+
+    it("Should set the right eventName and argumentName", () => {
```

Comment:
Are you wanting all `Should`'s capitalized in this case? Or is it a _don't care, its a convention not worth enforcing_?